protocol/x/clob/memclob/memclob.go

```diff
@@ -596,34 +540,30 @@ func (m *MemClobPriceTimePriority) PlaceOrder(
 		// off-chain message with the total filled size of the order equal to the size of the order.
 		// This is needed to account for the case where an order was partially matched, rewound, then was fully matched
 		// during uncrossing.
-		if m.generateOffchainUpdates {
-			if message, success := off_chain_updates.CreateOrderUpdateMessage(
-				ctx,
-				order.OrderId,
-				order.GetBaseQuantums(),
-			); success {
-				offchainUpdates.AddUpdateMessage(order.OrderId, message)
-			}
-		}
-		return orderSizeOptimisticallyFilledFromMatchingQuantums, takerOrderStatus.OrderStatus, offchainUpdates, nil
+		m.emitOrderUpdateEvent(
+			ctx,
+			order.OrderId,
+			order.GetBaseQuantums(),
+		)
+		m.emitOrderRemovalEvent(
+			ctx,
+			order.OrderId,
+			indexersharedtypes.OrderRemovalReason_ORDER_REMOVAL_REASON_FULLY_FILLED,
+			ocutypes.OrderRemoveV1_ORDER_REMOVAL_STATUS_BEST_EFFORT_CANCELED,
+		)
```

Comment:
This shouldn't be sent, this removal reason is only meant for orders that are confirmed to be fully filled from actual fills and not from the in-memory matching process of a node.

Reply:
Talked offline, concluded that this can be addressed in follow-up changes to Indexer to handle any change in behavior that leads to unwanted FE behavior.